inc/saiswitch.h

```diff
@@ -1995,6 +2237,62 @@ typedef void (*sai_switch_state_change_notification_fn)(
         _In_ sai_object_id_t switch_id,
         _In_ sai_switch_oper_status_t switch_oper_status);
 
+/**
+ * @brief Platform specific device register read access
+ *
+ * This API provides platform adaption in 2 ways
+ *
+ * 1# Provides platform adaption functionality to access device
+ * registers from driver. This is mandatory to pass as attribute to
+ * sai_create_switch when driver implementation does not support register access
+ * by device file system directly.
+ * 2# Provides device access API from ASIC SDK.
+ *
+ * @objects switch_id SAI_OBJECT_TYPE_SWITCH
+ *
+ * @param[in] platform_context Platform context information.
+ * @param[in] switch_id Switch Id
+ * @param[in] device_addr Device address(PHY/lane/port MDIO address)
+ * @param[in] start_reg_addr Starting register address to read
+ * @param[in] number_of_registers Number of consecutive registers to read
+ * @param[out] reg_val Register read values
+ */
+typedef sai_status_t (*sai_switch_register_read_fn)(
+        _In_ const void *platform_context,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t device_addr,
+        _In_ uint32_t start_reg_addr,
+        _In_ uint32_t number_of_registers,
+        _Out_ const uint32_t *reg_val);
+
+/**
+ * @brief Platform specific device register write access
+ *
+ * This API provides platform adaption in 2 ways
+ *
+ * 1# Provides platform adaption functionality to access device
+ * registers from driver. This is mandatory to pass as attribute to
+ * sai_create_switch when driver implementation does not support register access
+ * by device file system directly.
+ * 2# Provides device access API from ASIC SDK.
+ *
+ * @objects switch_id SAI_OBJECT_TYPE_SWITCH
+ *
+ * @param[in] platform_context Platform context information.
+ * @param[in] switch_id Switch Id
+ * @param[in] device_addr Device address(PHY/lane/port MDIO address)
+ * @param[in] start_reg_addr Starting register address to write
+ * @param[in] number_of_registers Number of consecutive registers to write
+ * @param[in] reg_val Register write values
+ */
+typedef sai_status_t (*sai_switch_register_write_fn)(
```

Comment:
could those functions be somehow transformed to attributes ?, if not then they will need some extra handling in sai metadata since they dont follow any currently supported format

Reply:
This can not be attribute, everything need to access PHY. Could you please help me to add special handling.

Reply:
easiest way is to add those 2 functions for ignore from check under style.pm, and look for function CheckFunctionNaming, you will see exceptions there (twice)